src/lib/device/device_manager.cpp

```diff
@@ -304,30 +304,46 @@ shared< VirtualDev > DeviceManager::create_vdev(vdev_parameters&& vparam) {
     shared< VirtualDev > vdev = m_vdev_create_cb(*vinfo, false /* load_existing */);
     m_vdevs[vdev_id] = vdev;
 
-    // Create initial chunk based on current size
-    if (vparam.num_chunks != 0) {
-        for (auto& pdev : pdevs) {
-            std::vector< uint32_t > chunk_ids;
-
-            // Create chunk ids for all chunks in each of these pdevs
-            for (uint32_t c{0}; c < vparam.num_chunks / pdevs.size(); ++c) {
-                auto chunk_id = m_chunk_id_bm.get_next_reset_bit(0u);
-                if (chunk_id == sisl::Bitset::npos) {
-                    throw std::out_of_range("System has no room for additional chunks");
-                }
-                m_chunk_id_bm.set_bit(chunk_id);
-                chunk_ids.push_back(chunk_id);
-            }
+    // different type might have different capacity, so we need to spread all the newly created chunks to all pdevs
+    // according to their capacity
 
-            // Create all chunks at one shot and add each one to the vdev
-            auto chunks = pdev->create_chunks(chunk_ids, vdev_id, vparam.chunk_size);
-            for (auto& chunk : chunks) {
-                vdev->add_chunk(chunk, true /* fresh_chunk */);
-                m_chunks[chunk->chunk_id()] = chunk;
-            }
+    // the total size of all pdevs of a certain type
+    auto total_type_size =
+        std::accumulate(pdevs.begin(), pdevs.end(), 0u, [](int r, const PhysicalDev* a) { return r + a->data_size(); });
+
+    uint32_t total_created_chunks{0};
+
+    for (auto& pdev : pdevs) {
+        if (total_created_chunks >= vparam.num_chunks) break;
+        std::vector< uint32_t > chunk_ids;
+
+        // the total number of chunks will be created in this pdev
+        auto total_chunk_num_in_pdev =
+            static_cast< uint32_t >(vparam.num_chunks * (pdev->data_size() / static_cast< float >(total_type_size)));
```

Comment:
nit: better to be `round_down(vparam.num_chunks * pdev->data_size() / total_type_size)`